tests/check/check_h26xsigned_auth.c

```diff
@@ -1661,17 +1661,58 @@ generate_and_set_private_key_on_camera_side(struct sv_setting settings, bool add
   sv_rc = signed_video_get_nalu_to_prepend(sv_camera, &nalu_to_prepend);
   ck_assert_int_eq(sv_rc, SV_OK);
   ck_assert(nalu_to_prepend.prepend_instruction == SIGNED_VIDEO_PREPEND_NOTHING);
+  ck_assert(tag_is_present(*sei, settings.codec, PUBLIC_KEY_TAG) == add_public_key_to_sei);
+  free(private_key);
+
+  return sv_camera;
+}
+
+static void validation_helper_function(signed_video_t *sv, nalu_list_item_t *sei, nalu_list_item_t *i_nalu, bool add_bad_public_key, bool late_public_key, signature_info_t *signature_info, bool any_public_key)
```

Comment:
By adding `codec` as input you can remove both `i_nalu` and `any_public_key`, see below.

You do not need `any_public_key` as input. You can compute it inside the function from the `sei` and with `late_public_key` as (changed name to `public_key_present`)
```
bool public_key_present = tag_is_present(*sei, codec, PUBLIC_KEY_TAG) || late_public_key;
```

Further, `late_public_key` can also be determined if you allow NULL pointer for `signature_info` if no public key should be added late
```
bool late_public_key = signature_info == NULL;
```

And finally, even `i_nalu`can be removed since it is always the same and can be generated internally as
```
nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
```

And last, I think `wrong_public_key` is a better name for `add_bad_public_key`.